# Notebook 01 — Data Collection

## Objectives
* Load the Digital Marketing Campaign dataset from the raw inputs folder
* Load the Marketing KPIs dataset from the raw inputs folder
* Perform an initial inspection of both datasets
* Confirm both datasets are present and valid for further analysis

## Inputs
* `inputs/datasets/raw/digital_marketing_campaign_dataset.csv`
* `inputs/datasets/raw/Marketing.csv`

## Outputs
* Confirmation that both datasets load correctly
* Initial shape, dtypes and missing value summary

## Notes
* AdvertisingPlatform and AdvertisingTool columns contain only confidential
  placeholder values and will be dropped in the cleaning notebook
* ConversionRate is a derived metric that risks data leakage and will be
  excluded from ML features

In [1]:
import os

# Navigate to project root from jupyter_notebooks folder
os.chdir(os.path.dirname(os.getcwd()))
print(f"Working directory: {os.getcwd()}")

Working directory: /workspaces/digital-marketing-conversion-predictor


---
## Import Libraries

In [2]:
import pandas as pd
import numpy as np

print(f"pandas version: {pd.__version__}")
print(f"numpy version: {np.__version__}")

pandas version: 2.2.3
numpy version: 2.2.4


---
## Load Dataset 1 — Digital Marketing Campaign

In [3]:
df1 = pd.read_csv('inputs/datasets/raw/digital_marketing_campaign_dataset.csv')
print(f"Shape: {df1.shape}")
df1.head()

Shape: (8000, 20)


,CustomerID,Age,Gender,Income,CampaignChannel,CampaignType,AdSpend,ClickThroughRate,ConversionRate,WebsiteVisits,PagesPerVisit,TimeOnSite,SocialShares,EmailOpens,EmailClicks,PreviousPurchases,LoyaltyPoints,AdvertisingPlatform,AdvertisingTool,Conversion
0,8000,56,Female,136912,Social Media,Awareness,6497.870068,0.043919,0.088031,0,2.399017,7.396803,19,6,9,4,688,IsConfid,ToolConfid,1
1,8001,69,Male,41760,Email,Retention,3898.668606,0.155725,0.182725,42,2.917138,5.352549,5,2,7,2,3459,IsConfid,ToolConfid,1
2,8002,46,Female,88456,PPC,Awareness,1546.429596,0.277490,0.076423,2,8.223619,13.794901,0,11,2,8,2337,IsConfid,ToolConfid,1
3,8003,32,Female,44085,PPC,Conversion,539.525936,0.137611,0.088004,47,4.540939,14.688363,89,2,2,0,2463,IsConfid,ToolConfid,1
4,8004,60,Female,83964,PPC,Conversion,1678.043573,0.252851,0.109940,0,2.046847,13.993370,6,6,6,8,4345,IsConfid,ToolConfid,1


In [4]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   CustomerID           8000 non-null   int64  
 1   Age                  8000 non-null   int64  
 2   Gender               8000 non-null   object 
 3   Income               8000 non-null   int64  
 4   CampaignChannel      8000 non-null   object 
 5   CampaignType         8000 non-null   object 
 6   AdSpend              8000 non-null   float64
 7   ClickThroughRate     8000 non-null   float64
 8   ConversionRate       8000 non-null   float64
 9   WebsiteVisits        8000 non-null   int64  
 10  PagesPerVisit        8000 non-null   float64
 11  TimeOnSite           8000 non-null   float64
 12  SocialShares         8000 non-null   int64  
 13  EmailOpens           8000 non-null   int64  
 14  EmailClicks          8000 non-null   int64  
 15  PreviousPurchases    8000 non-null   i

In [5]:
missing = df1.isnull().sum()
print("Missing values per column:")
print(missing[missing > 0] if missing.sum() > 0 else "No missing values found.")

Missing values per column:
No missing values found.


In [6]:
print("Target variable — Conversion:")
print(df1['Conversion'].value_counts())
print(f"\nConversion rate: {df1['Conversion'].mean():.2%}")

Target variable — Conversion:
Conversion
1    7012
0     988
Name: count, dtype: int64

Conversion rate: 87.65%


In [7]:
for col in ['AdvertisingPlatform', 'AdvertisingTool']:
    print(f"\n{col} unique values: {df1[col].unique()}")


AdvertisingPlatform unique values: ['IsConfid']

AdvertisingTool unique values: ['ToolConfid']


---
## Load Dataset 2 — Marketing KPIs

In [8]:
df2 = pd.read_csv('inputs/datasets/raw/Marketing.csv')
print(f"Shape: {df2.shape}")
df2.head()

Shape: (308, 11)


,id,c_date,campaign_name,category,campaign_id,impressions,mark_spent,clicks,leads,orders,revenue
0,1,2021-02-01,facebook_tier1,social,349043,148263,7307.37,1210,13,1,4981.0
1,2,2021-02-01,facebOOK_tier2,social,348934,220688,16300.20,1640,48,3,14962.0
2,3,2021-02-01,google_hot,search,89459845,22850,5221.60,457,9,1,7981.0
3,4,2021-02-01,google_wide,search,127823,147038,6037.00,1196,24,1,2114.0
4,5,2021-02-01,youtube_blogger,influencer,10934,225800,29962.20,2258,49,10,84490.0


In [9]:
print(df2.info())
print("\nMissing values:")
print(df2.isnull().sum())
print("\nCategories:")
print(df2['category'].value_counts())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 308 entries, 0 to 307
Data columns (total 11 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             308 non-null    int64  
 1   c_date         308 non-null    object 
 2   campaign_name  308 non-null    object 
 3   category       308 non-null    object 
 4   campaign_id    308 non-null    int64  
 5   impressions    308 non-null    int64  
 6   mark_spent     308 non-null    float64
 7   clicks         308 non-null    int64  
 8   leads          308 non-null    int64  
 9   orders         308 non-null    int64  
 10  revenue        308 non-null    float64
dtypes: float64(2), int64(6), object(3)
memory usage: 26.6+ KB
None

Missing values:
id               0
c_date           0
campaign_name    0
category         0
campaign_id      0
impressions      0
mark_spent       0
clicks           0
leads            0
orders           0
revenue          0
dtype: int64

Categories:
cat

---
## Conclusions

**Dataset 1 — Digital Marketing Campaign:**
- 8,000 records, 20 columns, no missing values
- Target `Conversion` is imbalanced: 87.65% positive class — SMOTE will be needed
- `AdvertisingPlatform` and `AdvertisingTool` contain only confidential placeholders — will be dropped
- `ConversionRate` is a derived metric that risks data leakage — will be excluded from ML features

**Dataset 2 — Marketing KPIs:**
- 308 records, 11 columns, no missing values
- 4 campaign categories: social (168), search (56), influencer (56), media (28)
- `campaign_name` has inconsistent capitalisation (e.g. facebOOK_tier2) — will be standardised in cleaning
- `c_date` is stored as object — will be parsed as datetime in cleaning

Both datasets are confirmed present and valid. Ready for Data Cleaning notebook.